sklearn/datasets/openml.py

```diff
@@ -0,0 +1,415 @@
+    elif len(results) == 1:
+        return results[0]
+    else:
+        return results
+
+
+def _determine_single_target_data_type(features_dict, target_column_name):
+    # determine the data type of the y array in case there is a single target
+    if not isinstance(target_column_name, string_types):
+        raise ValueError('target_column_name should be of string type, '
+                         'got: %s' % type(target_column_name))
+    if target_column_name not in features_dict:
+        raise KeyError('Could not find target_column_name={}')
+    feature = features_dict[target_column_name]
+    # note: we compare to a string, not boolean
+    if feature['is_ignore'] == 'true':
+        warn('target_column_name={} has flag is_ignore.'.format(
+            target_column_name))
+    if feature['is_row_identifier'] == 'true':
+        warn('target_column_name={} has flag is_row_identifier.'.format(
+            target_column_name))
+
+    if feature['data_type'] == "numeric":
+        return np.float64
+    else:
+        return object
+
+
+def _determine_multi_target_data_type(features_dict, target_column_names):
+    # determine the data type of the y array in case there are multiple targets
+    # (throws an error if these targets do not comply with sklearn support)
+    if not isinstance(target_column_names, list):
+        raise ValueError('target_column_name should be list, '
+                         'got: %s' % type(target_column_names))
+    found_types = set()
+    for target_column_name in target_column_names:
+        if target_column_name not in features_dict:
+            raise KeyError('Could not find target_column_name={}')
+        if features_dict[target_column_name]['data_type'] == "numeric":
+            found_types.add(np.float64)
+        else:
+            found_types.add(object)
+
+        # note: we compare to a string, not boolean
+        if features_dict[target_column_name]['is_ignore'] == 'true':
+            warn('target_column_name={} has flag is_ignore.'.format(
+                target_column_name))
+        if features_dict[target_column_name]['is_row_identifier'] == 'true':
+            warn('target_column_name={} has flag is_row_identifier.'.format(
+                target_column_name))
+    if len(found_types) != 1:
+        raise ValueError('Can only handle homogeneous multi-target datasets, '
+                         'i.e., all targets are either numeric or '
+                         'categorical.')
+    return list(found_types)[0]
+
+
+def fetch_openml(name=None, version='active', data_id=None, data_home=None,
+                 target_column_name='default-target', cache=True):
+    """Fetch dataset from openml by name or dataset id.
+
+    Datasets are uniquely identified by either an integer ID or by a
+    combination of name and version (i.e. there might be multiple
+    versions of the 'iris' dataset). Please give either name or id
+    (not both). In case a name is given, a version can also be
```

Comment:
perhaps: can -> should

Reply:
Although I am not a native speaker, to me it seems that can is more open-ended than should. Note that it is fine to not provide the version, providing it is just a bit better. What do you think?